### PR TITLE
Refactor git log analysis and add more testing

### PR DESCRIPTION
**Pull request by maxjeffos**
- [x] Ready for review
- [X] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [x] Reviewed by Snyk internal team

#### What does this PR do?
- minor refactoring of src/lib/monitor/dev-count-analysis.ts to facility improved testing
- add additional testing to ensure merge commits are not counted in CLI dev count feature

#### Where should the reviewer start?
Reviewing this very small PR: https://github.com/snyk/snyk/pull/1408

#### Any background context you want to provide?
I should have included more testing in the last PR. This adds testing to make sure that adding `no-merges` to the git log command works as intended.

#### What are the relevant tickets?
HAMMER-156

### CODE REVIEW

**Comment by maxjeffos** [comment]
Should wait till https://github.com/snyk/snyk/pull/1410 is merged and then rebase this branch to master to make sure that the Jest tests are working on Windows.

**Review by darscan — APPROVED** [review]

File: src/cli/commands/monitor/index.ts [hunk 1]
```diff
@@ -93,22 +86,10 @@ async function monitor(...args0: MethodArgs): Promise<any> {
...
-        CONTRIBUTING_DEVELOPER_PERIOD_DAYS,
-      );
-      const gitLogResults = await runGitLog(
-        timestampStartOfContributingDeveloperPeriod,
-        repoPath,
-        execShell,
-      );
-      const stats: GitRepoCommitStats = parseGitLog(gitLogResults);
-      contributors = stats.getRepoContributors();
+      contributors = await countContributors();
```
Comment (darscan) [inline src/cli/commands/monitor/index.ts:92]: Much cleaner!

File: src/lib/monitor/dev-count-analysis.ts [hunk 2]
```diff
@@ -140,13 +160,18 @@ export function getTimestampStartOfContributingDevTimeframe(
...
 }
 
-export async function runGitLog(
+export function buildGitLogCommandLine(
   timestampEpochSecondsStartOfPeriod: number,
+): string {
+  return `git --no-pager log --no-merges --pretty=tformat:"%H${SERIOUS_DELIMITER}%an${SERIOUS_DELIMITER}%ae${SERIOUS_DELIMITER}%aI" --after="${timestampEpochSecondsStartOfPeriod}"`;
+}
+
+export async function runGitLog(
```
Comment (darscan) [inline src/lib/monitor/dev-count-analysis.ts:169]: This is much worse though. Both functions are anaemic and don't make sense alone. `runGitLog` has become especially brittle and nonsensical, as you can call it with any string as its firs argument: `runGitLog('rm -rf')`.

File: src/lib/types.ts [hunk 3]
```diff
@@ -29,7 +29,7 @@ export interface WizardOptions {
   newPolicy: boolean;
 }
 
-export interface Contributors {
+export interface Contributor {
```
Comment (darscan) [inline src/lib/types.ts:32]: Nice catch

**Review by JackuB — APPROVED** [review]
Left some comments, but approving as a whole :)

File: src/cli/commands/monitor/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (darscan) [reply src/cli/commands/monitor/index.ts:92]: Haha! Good point :D

File: src/lib/monitor/dev-count-analysis.ts (hunk 2 above)
Comment (darscan) [reply src/lib/monitor/dev-count-analysis.ts:169]: BTW, @maxjeffos knows how I feel about this part of the refactor, and I'm not blocking the PR or anything, I'm just venting my feels ☠️

File: src/cli/commands/monitor/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (maxjeffos) [reply src/cli/commands/monitor/index.ts]: Fixed

File: src/lib/monitor/dev-count-analysis.ts (hunk 2 above)
Comment (maxjeffos) [reply src/lib/monitor/dev-count-analysis.ts]: I restored the original form and removed the test that it facilitated.

**Pull request merged**